### PR TITLE
Ot153 38 newsletter form

### PR DESCRIPTION
**Pull request by FacundoSaldivia**
## JIRA
[OT153-038](https://alkemy-labs.atlassian.net/browse/OT153-038)

## Short description

Input of newsletter for footer. Saves the email of the user on LocalStorage when subscribed to the newsletter

## Screenshot or Video

https://user-images.githubusercontent.com/50809738/155554307-190f4fec-f5d2-449a-892b-f90c8ab26c96.mp4

## Teammate collaborator

- [ ] Someone from the team collaborated (**_if so, please tag them after this line_**).

### Sanity Check Steps

- [x] All console logs have been removed.
- [x] This change was tested on my localhost.
- [ ] This change depends for another ticket.
- [x] All linked JIRA acceptance criteria are satisfied.
- [ ] Does this change require environment variables.

### CODE REVIEW

File: src/app/features/features.module.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ import { ErrorComponent } from "./public/components/donations/error/error.compon
     ErrorComponent,
     ActivitiesCardComponent,
     ActivitiesComponent,
+    NewsletterFormComponent,
```
Comment (RicardoMaldonado93) [inline src/app/features/features.module.ts:43]: Newsletter es un componente que va en el CoreModule

File: src/app/features/public/components/footer/newsletter-form/newsletter-form.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div class="wrapper">
```
Comment (RicardoMaldonado93) [inline src/app/features/public/components/footer/newsletter-form/newsletter-form.component.html:1]: mover el componente al CoreModule/components y darle un estilo mas lindo, te dejo uno a modo de ejemplo
( este componente va dentro del footer ! )
image

File: src/app/features/public/components/footer/newsletter-form/newsletter-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  ngOnInit(): void {
+    this.checkRegistred();
+  }
+  submit(): void {
+    localStorage.setItem("user_email", this.form.value.email);
+    this.checkRegistred();
+  }
+
+  get email() {
```
Comment (RicardoMaldonado93) [inline src/app/features/public/components/footer/newsletter-form/newsletter-form.component.ts:25]: remover este y usar el utils que se encuentra en el Core

File: src/app/features/public/components/footer/newsletter-form/newsletter-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  selector: "app-newsletter-form",
+  templateUrl: "./newsletter-form.component.html",
+  styleUrls: ["./newsletter-form.component.scss"],
+})
+export class NewsletterFormComponent implements OnInit {
+  constructor() {}
+  form: FormGroup = new FormGroup({
+    email: new FormControl("", [Validators.required, emailValidator()]),
+  });
+  user_email: string = "";
```
Comment (RicardoMaldonado93) [inline src/app/features/public/components/footer/newsletter-form/newsletter-form.component.ts:15]: las propiedades siempre van primero , luego el constructor y luego el resto de las funciones

**Review by RicardoMaldonado93 — COMMENTED** [review]
Please fix the observations to continues with the PR

File: src/app/features/features.module.ts (hunk 1 above)
Comment (FacundoSaldivia) [reply src/app/features/features.module.ts:43]: Una consulta, si muevo el componente a core, tambien deberia importar en coreModule el reactive forms ?

Comment (FacundoSaldivia) [reply src/app/features/features.module.ts:43]: Una consulta, deberia importar reactive forms en core module ?

Comment (RicardoMaldonado93) [reply src/app/features/features.module.ts:43]: Si deberias de hacerlo

**Comment by RicardoMaldonado93** [comment]
unused branch

**Pull request closed without merging**